tests/test_e2e.py

```diff
@@ -0,0 +1,148 @@
+import lightgbm as lgb
+import numpy as np
+import pytest
+
+try:
+    # neptune-client=0.9.0+ package structure
+    from neptune.new import init_run
+    from neptune.new.integrations.lightgbm import (
+        NeptuneCallback,
+        create_booster_summary,
+    )
+except ImportError:
+    # neptune-client>=1.0.0 package structure
+    from neptune import init_run
+    from neptune.integrations.lightgbm import (
+        NeptuneCallback,
+        create_booster_summary,
+    )
+
+args = (
+    (True, True, True, True, True),
+    (False, False, False, False, False),
+)
+
+
+@pytest.mark.parametrize("args", args)
+def test_e2e(dataset, args):
+    # Since these all arguments are independent of each other,
+    # we don't check all combinations.
+    log_importances, log_confusion_matrix, log_pickled_booster, log_trees, log_trees_as_dataframe = args
+    # Start a run
+    run = init_run()
+
+    # Create a NeptuneCallback instance
+    neptune_callback = NeptuneCallback(run=run)
+
+    X_train, X_test, y_train, y_test = dataset
+
+    lgb_train = lgb.Dataset(X_train, y_train)
+
+    # Define model parameters
+    params = {
+        "boosting_type": "gbdt",
+        "objective": "multiclass",
+        "num_class": 3,
+    }
+
+    # Train the model
+    gbm = lgb.train(
+        params,
+        lgb_train,
+        callbacks=[neptune_callback],
+    )
+
+    y_pred = np.argmax(gbm.predict(X_test), axis=1)
+
+    run["summary"] = create_booster_summary(
+        booster=gbm,
+        log_trees=log_trees,
+        log_trees_as_dataframe=log_trees_as_dataframe,
+        list_trees=[0, 1],
+        log_confusion_matrix=log_confusion_matrix,
+        log_importances=log_importances,
+        log_pickled_booster=log_pickled_booster,
+        y_pred=y_pred,
+        y_true=y_test,
+    )
+
+    run.wait()
+    validate_results(
+        run,
+        log_importances,
+        log_confusion_matrix,
+        log_pickled_booster,
+        log_trees,
+        log_trees_as_dataframe,
+        base_namespace="",
+    )
+
+
+def test_e2e_using_namespace(dataset):
```

Comment:
Maybe we could also test it using a handler?

```
run = init_run()
handler = run["subdirectory"]
neptune_callback = NeptuneCallback(run=handler, base_namespace="training")
```

Reply:
Makes sense. Thank you!